src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Services/SyntaxFacts/AbstractSyntaxFacts.cs

```diff
@@ -126,8 +126,13 @@ public bool IsOnSingleLine(SyntaxNode node, bool fullSpan)
                     var index = 0;
                     foreach (var child in childNodesAndTokens.Reverse())
                     {
-                        var first = index == 0;
-                        var last = index == childCount - 1;
+                        // Since we're walking the children in reverse, if we're on hte 0th item,
+                        // that's the last child.
+                        var last = index == 0;
+
+                        // Once we get all the way to the end of hte reversed list, we're actually
+                        // on the first.
+                        var first = index == childCount - 1;
```

Comment:
this logic bug caused us to *not* consider the `{ }` in `public C() { }` ask being on a single line.  because of that, we didn't properly try to wrap it in the calling feature.